gm8emulator/src/gml/kernel.rs

```diff
@@ -472,19 +472,53 @@ impl Game {
         expect_args!(args, [int]).map(|c| (c / 256 / 256) % 256).map(Value::from)
     }
 
-    pub fn color_get_hue(&mut self, _context: &mut Context, _args: &[Value]) -> gml::Result<Value> {
-        // Expected arg count: 1
-        unimplemented!("Called unimplemented kernel function color_get_hue")
+    pub fn color_get_hue(&mut self, _context: &mut Context, args: &[Value]) -> gml::Result<Value> {
+        let c = expect_args!(args, [int])?;
+        let (h, _s, _v) = self.rgb_to_hsv(c);
+        Ok(h.into())
     }
 
-    pub fn color_get_saturation(&mut self, _context: &mut Context, _args: &[Value]) -> gml::Result<Value> {
-        // Expected arg count: 1
-        unimplemented!("Called unimplemented kernel function color_get_saturation")
+    pub fn color_get_saturation(&mut self, _context: &mut Context, args: &[Value]) -> gml::Result<Value> {
+        let c = expect_args!(args, [int])?;
+        let (_h, s, _v) = self.rgb_to_hsv(c);
+        Ok(s.into())
     }
 
-    pub fn color_get_value(&mut self, _context: &mut Context, _args: &[Value]) -> gml::Result<Value> {
-        // Expected arg count: 1
-        unimplemented!("Called unimplemented kernel function color_get_value")
+    pub fn color_get_value(&mut self, _context: &mut Context, args: &[Value]) -> gml::Result<Value> {
+        let c = expect_args!(args, [int])?;
+        let (_h, _s, v) = self.rgb_to_hsv(c);
+        Ok(v.into())
+    }
+
+    pub fn rgb_to_hsv(&mut self, color: i32) -> (i32, i32, i32) {
+        let (r, g, b) = (Real::from(color & 255), Real::from((color >> 8) & 255), Real::from((color >> 16) & 255));
```

Comment:
A constant like `0xFF` (preferred) or `0b1111_1111` would indicate intent better here

Reply:
I converted the integers to hex where they were related to anything to do with binary/data sizes.

I left the 120, 240, 360 in decimal for readability, and those numbers show that either degrees or a circle is involved.